filewrangle.py

```diff
@@ -0,0 +1,147 @@
+       #copy_to(from_dir_filename, post_dir, post_dir_filename)
+       move_to(from_dir_filename, post_dir, post_dir_filename) 
+       
+  end_time = datetime.now()
+  
+  #total_time_to_upload = relativedelta(end_time, start_time)
+  total_time_to_upload = end_time - start_time
+    
+  print ("%s ***Processing complete***  File count: %d Size: %d kb Total time to upload:   %s seconds" % (datetime.now().strftime('%Y-%m-%d %H:%M:%S'), i, total_file_size, total_time_to_upload.seconds))
+  return 
+  
+def get_key(filename):
+  object_key = ''  
+  keyparts = filename.split("_")  
+  if len(keyparts) > 1:
+     if keyparts[0] == "ServiceName":
+        object_key = get_service_format_key(keyparts)
+     elif keyparts[1] == "ApplicationName":
+          object_key = get_app_format_key(keyparts) 
+  #skip anything else     
+  return object_key
+  
+def get_service_format_key(keyparts):
+# return type server date and datetime
+  return keyparts[0] + "_" + keyparts[2] + "_" + keyparts[1] + ".log.gz"
+  
+def get_app_format_key(keyparts):
+  # return type server date and datetime
+  return keyparts[1] + "_" + keyparts[0] + "_" + keyparts[2] + ".log.gz"
+  
+def get_doc_format_key(keyparts):
+  # return type server date and datetime
+  return  keyparts[0] 
+  
+def copy_to(from_file_dir, post_dir, post_dir_file): 
+    if not os.path.exists(post_dir):   
+        os.mkdir(post_dir)   
+
+    shutil.copy(from_file_dir, post_dir_file)
+    return
+
+def move_to(from_file_dir, post_dir, post_dir_file): 
+    if not os.path.exists(post_dir):   
+        os.mkdir(post_dir)   
+
+    shutil.move(from_file_dir, post_dir_file)
+    return 
+    
+def main():
+  print ('--------------------------------------------------------------------------')
+  result = []
+  # This basic command line argument parsing code is provided. 
+
+  # Make a list of command line arguments, omitting the [0] element
+  # which is the script itself.
+  args = sys.argv[1:]
+  if not args:
+    print ("usage: --[fromdir] dir --[postdir] dir --[bucketname] bucketname")
+    sys.exit(1)
+
+  if len(args) <> 6:
+    print ("error: provide from and post processing directories")
+    sys.exit(1)
+
+  # dir is set from command line
+  # or left as the empty string. 
+  fromdir = ''
+  if args[0] == '--fromdir':
+    fromdir = args[1]
+    del args[0:2]
+        
+  postdir = ''
+  if args[0] == '--postdir':
+    postdir = args[1]
+    del args[0:2]
+          
+  bucketname = ''
+  if args[0] == '--bucketname':
+    bucketname = args[1]
+    del args[0:2]
+        
+  if fromdir and postdir and bucketname:
+    print ("Source file directory: %s " % fromdir)   
+    print ("Post processing directory: %s" % postdir)
+    print ("S3 bucketname: %s" % bucketname)
+    print ("  ")
+    print ("  " )
+    process_files(fromdir, postdir, bucketname)
+  else:
+    print ('\n'.join(result))
```

Comment:
'result' is declared as an array in line 105, but then never filled, and then it's printed (empty?)  when there are no arguments on the command line.  I don't know what function 'result' is serving.